discovery/gossiper.go

```diff
@@ -1782,8 +1947,13 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 		// making decisions based on this DB state, before it
 		// writes to the DB.
 		d.channelMtx.Lock(msg.ShortChannelID.ToUint64())
-		defer d.channelMtx.Unlock(msg.ShortChannelID.ToUint64())
-		if err := d.cfg.Router.AddEdge(edge, schedulerOp...); err != nil {
+		err := d.cfg.Router.AddEdge(edge, schedulerOp...)
+		if err == nil {
```

Comment:
nit: not sure how the logic is easier to follow. What we have right now or if we do

```
err := d.cfg.Router.AddEdge(edge, schedulerOp...)
if err != nil {
    defer + logic
}
// unlock and continue
d.channelMtx.Unlock(msg.ShortChannelID.ToUint64())
```

Reply:
Yeah neither is good imo. Switched to your suggestion as it involves less indentation. But honestly, this package needs to be refactored, this function has like 1000 lines...